### PR TITLE
Upgrade go to 1.18 to resolve security vulnerability CVE-2022-27191

### PR DESCRIPTION
**Pull request by dbrenecki**
Bump go to 1.18 to resolve vulnerability - [CVE-2022-27191](https://nvd.nist.gov/vuln/detail/CVE-2022-27191)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
+FROM golang:1.18.0 AS build
 ENV PROJECT grpc_health_probe
 WORKDIR /src/$PROJECT
 COPY go.mod go.sum ./
 RUN go mod download
 COPY . .
 RUN CGO_ENABLED=0 go install -a -tags netgo -ldflags=-w
 
-FROM alpine:3.8
+FROM alpine:3.15.4
```
Comment (ahmetb) [inline Dockerfile:9]: do you mind just doing :3 or :3.15 here?

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
-FROM golang:1.13 AS build
+FROM golang:1.18.0 AS build
```
Comment (ahmetb) [inline Dockerfile:1]: do you mind doing just :1.18 here? that way we can get the patches.

Comment (breneckd) [reply Dockerfile]: Updated

File: Dockerfile (hunk 1 above)
Comment (breneckd) [reply Dockerfile]: Updated

**Comment by chitran96** [comment]
hey @dbrenecki, what's we waiting for? Can we merge this PR and release new version of `grpc-health-probe` to address [CVE-2022-27191](https://nvd.nist.gov/vuln/detail/CVE-2022-27191)? Thanks :D

**Comment by breneckd** [comment]
> hey @dbrenecki @ahmetb, what's we waiting for? Can we merge this PR and release new version of `grpc-health-probe` to address [CVE-2022-27191](https://nvd.nist.gov/vuln/detail/CVE-2022-27191)? Thanks :D

Waiting on ahmetb I do not have permissions to merge.

**Pull request merged**